### PR TITLE
Story/io 334/construction program report

### PR DESCRIPTION
**Pull request by EngbergJonas**
- Download construction program report in pdf format with react-pdf/renderer
- Download construction program report in csv format with react-csv

### CODE REVIEW

File: src/utils/calculations.ts [hunk 1]
```diff
@@ -169,8 +169,19 @@ export const calculateProjectRowSums = (project: IProject): IProjectSums => {
...
     availableFrameBudget: formatNumber(availableFrameBudget),
-    costEstimateBudget: formatNumber(parseInt(costForecast ?? '0') - spentBudget),
+    costEstimateBudget: formatNumber(parseInt(costForecast ?? '0') - parseInt(spentBudget)),
   };
 };
 
 export const calcPercentage = (value: number, total: number) => Math.round((value / total) * 100);
+
+export const keurToMillion = (value?: string | null) => {
+  if (value) {
```
Comment (kgunga) [inline src/utils/calculations.ts:179]: if (!value) return '0,0';

const valueAsNumber = parseFloat(value);
    const millionValue = (valueAsNumber / 1000).toFixed(1);

    return millionValue.toString().replace('.', ',');

**Review by kgunga — CHANGES_REQUESTED** [review]

File: src/components/Report/DownloadCsvButton.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  divisions: Array<ILocation>;
+  classes: IClassHierarchy;
+}
+
+const getConstructionProgramReportData = async (
+  classes: IClassHierarchy,
+  divisions: Array<ILocation>,
+  t: TFunction<'translation', undefined>,
+): Promise<Array<IConstructionProgramCsvRow>> => {
+  const projects = await getProjectsWithParams({
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Report/DownloadCsvButton.tsx]: fixed

File: src/components/Report/DownloadCsvButton.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    programmed: false,
+    params: 'overMillion=true',
+  })
+    .then((res) => res.results)
+    .catch((e) => {
+      console.log('Error downloading CSV: ', e);
+      return [];
+    });
+
+  if (projects) {
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Report/DownloadCsvButton.tsx]: fixed

File: src/components/Report/DownloadCsvButton.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+/**
+ * We're using pdf-react to create pdf's.
+ *
+ * The styles are a bit funky since pdf-react doesn't support grid or table.
+ */
+const DownloadCsvButton: FC<IDownloadCsvButtonProps> = ({ type, divisions, classes }) => {
+  const { t } = useTranslation();
+  const [csvData, setCsvData] = useState<Array<IConstructionProgramCsvRow>>([]);
+
+  const downloadIcon = useMemo(() => <IconDownload />, []);
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Report/DownloadCsvButton.tsx]: fixed

File: src/components/Report/DownloadCsvButton.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    }
+  }, [csvData]);
+
+  const getCsvData = useCallback(async () => {
+    switch (type) {
+      case 'constructionProgram':
+        setCsvData(await getConstructionProgramReportData(classes, divisions, t));
+        break;
+      default:
+        // implemented later...
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Report/DownloadCsvButton.tsx]: fixed

File: src/components/Report/DownloadCsvButton.tsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        >
+          {t('downloadXlsx', { name: t(`report.${type}.documentName`) })}
+        </Button>
+      </div>
+      {csvData.length > 0 ? <CSVDownload data={csvData} target="_blank" /> : undefined}
+    </>
+  );
+};
+
+export default DownloadCsvButton;
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Report/DownloadCsvButton.tsx]: fixed

File: src/components/Report/DownloadPdfButton.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+      iconLeft={downloadIcon}
+      onClick={() => downloadPdf()}
+      disabled={type !== 'constructionProgram'}
+    >
+      {t('downloadPdf', { name: documentName })}
+    </Button>
+  );
+};
+
+export default DownloadPdfButton;
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Report/DownloadPdfButton.tsx]: fixed

File: src/utils/calculations.ts (hunk 1 above)
Comment (EngbergJonas) [reply src/utils/calculations.ts]: fixed

**Review by kgunga — APPROVED** [review]
👍

**Pull request merged**